com.unity.robotics.urdf-importer/Runtime/RuntimeImport/RuntimeURDFImporter.cs

```diff
@@ -0,0 +1,138 @@
+        {
+            isLoading = false;
+            yield break;
+        }
+
+        // clear the existing robot to avoid collision
+        if (clearOnLoad && currentRobot != null)
+        {
+            currentRobot.SetActive(false);
+            Destroy(currentRobot);
+        }
+        yield return null;
+
+        ImportSettings settings = new ImportSettings
+        {
+            choosenAxis = ImportSettings.axisType.yAxis,
+            convexMethod = useVHACD ? ImportSettings.convexDecomposer.vHACD : ImportSettings.convexDecomposer.unity,
+        };
+
+        GameObject robotObject = null;
+        if (showProgress) 
+        {
+            IEnumerator<GameObject> createRobot = UrdfRobotExtensions.Create(urdfFilepath, settings, showProgress, true);
+            yield return createRobot;
+            robotObject = createRobot.Current;
+        }
+        else
+        {
+            robotObject = UrdfRobotExtensions.CreateRuntime(urdfFilepath, settings);
+        }
+
+        if (robotObject != null && robotObject.transform != null) 
+        {
+            robotObject.transform.SetParent(transform);
+            SetControllerParameters(robotObject);
+            Debug.Log("Successfully Loaded URDF" + robotObject.name);
+            //robotObject.SetActive(true);
+        }
+        currentRobot = robotObject;
+        isLoading = false;
+    }
+
+    void SetControllerParameters(GameObject robot)
+    {
+        if (setImmovableLink) 
+        {
+            Transform baseNode = robot.transform.FirstChildOrDefault(x => x.name == immovableLinkName);
+            if (baseNode && baseNode.TryGetComponent<ArticulationBody>(out ArticulationBody baseNodeAB)) 
+            {
+                baseNodeAB.immovable = true;
+            }
+        }
+
+        if (robot.TryGetComponent<Controller>(out Controller controller))
+        {
+            controller.stiffness = controllerStiffness;
+            controller.damping = controllerDamping;
+            controller.forceLimit = controllerForceLimit;
+            controller.speed = controllerSpeed;
+            controller.acceleration = controllerAcceleration;
+            controller.enabled = true;
+        }
+    }
+
+    void OnGUI()
+    {
+        urdfFilepath = GUI.TextField(new Rect(10, 50, 500, 25), urdfFilepath);
+        setImmovableLink = GUI.Toggle(new Rect(10, 75, 200, 25), setImmovableLink, "set immovable link");
+        if (setImmovableLink)
+        {
+            immovableLinkName = GUI.TextField(new Rect(220, 75, 200, 25), immovableLinkName);
+        }
+
+        useVHACD = GUI.Toggle(new Rect(10, 100, 200, 25), useVHACD, "Use vHACD");
+        showProgress = GUI.Toggle(new Rect(10, 125, 200, 25), showProgress, "Show Progress (experimental)");
+        if (!isLoading && GUI.Button(new Rect(530, 50, 150, 25), "Load UDRF File"))
+        {
+            StartCoroutine(LoadURDF());
+        }
+    }    
+}
+
+public static class TransformEx
```

Comment:
We can move this in the file Transform extensions.

Reply:
Done. 1f0c2fb